server/cluster/coordinator.go

```diff
@@ -157,6 +162,46 @@ func (c *coordinator) patrolRegions() {
 	}
 }
 
+// checkSuspectKeyRanges would pop one suspect key range group
+// The regions of new version key range and old version key range would be placed into
+// the suspect regions map
+func (c *coordinator) checkSuspectKeyRanges() {
+	_, keyRange, success := c.cluster.PopOneSuspectKeyRange()
+	if !success {
+		return
+	}
+	limit := 1024
+	regions := c.cluster.ScanRegions(keyRange[0], keyRange[1], limit)
+	if len(regions) < 1 {
+		return
+	}
+	// if the size equals limit which means there probably existed the rest regions.
+	// so we search the rest key range and put these key range into suspect keyRanges again.
+	if len(regions) == limit {
+		// search the biggest start Key in given regions
+		var biggestStartKey []byte
```

Comment:
I think you can use the last region directly.
Also, no need to check `if len(regions) == limit`, compare last region's endKey with `keyRange[1]` is better because `len(regions) == limit` does not necessarily mean there are more regions.

Reply:
updated.